src/libraries/System.Memory/tests/ReadOnlySpan/Count.byte.cs

```diff
@@ -0,0 +1,313 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Numerics;
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static partial class ReadOnlySpanTests
+    {
+        [Fact]
+        public static void ZeroLengthCount_Byte()
+        {
+            ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(Array.Empty<byte>());
+
+            int count = span.Count<byte>(0);
+            Assert.Equal(0, count);
+        }
+        
+        [Fact]
+        public static void ZeroLengthCount_RosByte()
+        {
+            ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(Array.Empty<byte>());
+
+            int count = span.Count<byte>(new ReadOnlySpan<byte>(new byte[] { 0, 0 }));
+            Assert.Equal(0, count);
+        }
+        
+        [Fact]
+        public static void DefaultFilledCount_Byte()
+        {
+            for (int length = 0; length <= byte.MaxValue; length++)
```

Comment:
We don't need to test all 256 byte values.  You could choose a lower max and/or just choose a few specific values to validate various border conditions.